### PR TITLE
feat: update dependencies, including latest proofs

### PR DESCRIPTION
**Pull request by cryptonemo**
fix: add back in rust native target building

### CODE REVIEW

**Comment by dignifiedquire** [comment]
@Kubuxu @magik6k  important note: using this in lotus, requires to upgrade blst to `0.3.2`

File: .circleci/config.yml
Comment (dignifiedquire) [inline .circleci/config.yml:296]: @cryptonemo I don't think this is a good idea, this will specialize to the CPU that circle ci has, ending up in randomly hard to detect issues. If folks want to build against native, they will need to build for themselves.

**Review by dignifiedquire — CHANGES_REQUESTED** [review]

File: .circleci/config.yml
Comment (cryptonemo) [reply .circleci/config.yml:296]: These builds aren't published. Maybe better to remove the CI jobs?

Comment (dignifiedquire) [reply .circleci/config.yml:296]: ah right, I always get confused with this CI job

File: rust/Cargo.lock
Comment (dignifiedquire) [inline rust/Cargo.lock:29]: @cryptonemo this file should not be removed, we need to have a fixed Cargo.lock

Comment (cryptonemo) [reply rust/Cargo.lock:29]: I'm not following.  Cargo.lock was updated to the latest deps, not removed.

Comment (dignifiedquire) [reply rust/Cargo.lock:29]: damn, I am not on my game today, there where so many deletions and got confused

**Review by dignifiedquire — APPROVED** [review]

**Comment by cryptonemo** [comment]
@dignifiedquire Thanks for the review, but was hoping @magik6k or @Kubuxu could also approve if it looks good/tests well.

**Comment by dignifiedquire** [comment]
Do not merge this PR yet, the blst update is likely to break in a way need to first figure out to work around

**Comment by cryptonemo** [comment]
> Do not merge this PR yet, the blst update is likely to break in a way need to first figure out to work around

Right.  It won't be merged until all is resolved there.

**Comment by cryptonemo** [comment]
No longer needed

**Pull request closed without merging**
